File: app/android/src/main/kotlin/com/personal/image_toolkit/ui/screen/ConvertScreen.kt
// ui/screen/ConvertScreen.kt
package com.example.imagetoolkit.ui.screen

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.imagetoolkit.ui.components.FileInput
import com.example.imagetoolkit.ui.components.FormatSelector
import android.widget.Toast

@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun ConvertScreen() {
    var outputFormat by remember { mutableStateOf("png") }
    var inputPath by remember { mutableStateOf("") }
    var outputPath by remember { mutableStateOf("") }
    var selectedFormats by remember { mutableStateOf(setOf<String>()) }
    val context = LocalContext.current
    val allFormats = listOf("jpg", "png", "bmp", "gif", "webp", "tiff")

    LazyColumn(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        item { Text("Convert Image Format", style = MaterialTheme.typography.headlineSmall) }
        item {
            OutlinedTextField(
                value = outputFormat,
                onValueChange = { outputFormat = it },
                label = { Text("Output Format") },
                modifier = Modifier.fillMaxWidth()
            )
        }
        item {
            FileInput("Input Path (File or Dir)", inputPath) { inputPath = it }
        }
        item {
            com.example.imagetoolkit.ui.components.SectionCard("Output Path (Optional)") {
                FileInput("Output Path", outputPath) { outputPath = it }
            }
        }
        item {
            FormatSelector(
                title = "Input Formats (if input is dir)",
                formats = allFormats,
                selectedFormats = selectedFormats,
                onFormatToggle = { format ->
                    selectedFormats = if (selectedFormats.contains(format))
                        selectedFormats - format else selectedFormats + format
                }
            )
        }
        item {
            Button(
                onClick = {
                    val msg = "Running Convert: \n- Format: $outputFormat\n- Input: $inputPath"
                    Toast.makeText(context, msg, Toast.LENGTH_LONG).show()
                },
                modifier = Modifier.fillMaxWidth()
            ) {
                Icon(Icons.Default.PlayArrow, contentDescription = null, modifier = Modifier.padding(end = 8.dp))
                Text("Run Conversion", fontSize = 16.sp)
            }
        }
    }
}

File: app/ios/image_toolkit/ui/screen/ConvertScreen.swift
import SwiftUI

/// The image-format conversion screen.
///
/// Lets the user specify an output format, an input file or directory path,
/// an optional output path, and a set of input formats to filter on when the
/// input is a directory. Pressing "Run Conversion" calls the backend via the
/// Tauri IPC bridge (stubbed here as an alert for development).
///
/// Mirrors `ConvertTab` in the PySide6 desktop GUI (`gui/src/tabs/core/convert_tab.py`).
struct ConvertScreen: View {
    @State private var outputFormat: String = "png"
    @State private var inputPath: String = ""
    @State private var outputPath: String = ""
    @State private var selectedFormats: Set<String> = []
    @State private var showingAlert = false
    @State private var alertMessage = ""

    let allFormats = ["jpg", "png", "bmp", "gif", "webp", "tiff"]

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                Text("Convert Image Format")
                    .font(AppTheme.Typography.headlineSmall)
                
                VStack(alignment: .leading, spacing: 8) {
                    Text("Output Format")
                        .font(.caption)
                        .foregroundColor(.gray)
                    TextField("png", text: $outputFormat)
                        .textFieldStyle(RoundedBorderTextFieldStyle())
                }
                
                FileInput(label: "Input Path (File or Dir)", path: $inputPath)
                
                SectionCard(title: "Output Path (Optional)") {
                    FileInput(label: "Output Path", path: $outputPath)
                }
                
                FormatSelector(
                    title: "Input Formats (if input is dir)",
                    formats: allFormats,
                    selectedFormats: $selectedFormats
                )
                
                Button(action: {
                    alertMessage = "Running Convert: \n- Format: \(outputFormat)\n- Input: \(inputPath)"
                    showingAlert = true
                }) {
                    HStack {
                        Image(systemName: "play.fill")
                        Text("Run Conversion")
                    }
                    .frame(maxWidth: .infinity)
                    .padding()
                    .background(AppTheme.primary)
                    .foregroundColor(.white)
                    .cornerRadius(8)
                }
            }
            .padding(16)
        }
        .alert("Convert Action", isPresented: $showingAlert) {
            Button("OK", role: .cancel) { }
        } message: {
            Text(alertMessage)
        }
    }
}